src/django_components/context.py

```diff
@@ -110,15 +133,31 @@ def set_outer_root_context(context: Context, outer_ctx: Optional[Context]) -> No
     We pass through this context to allow to configure how slot fills should be
     rendered using the `SLOT_CONTEXT_BEHAVIOR` setting.
     """
-    if outer_ctx and len(outer_ctx.dicts) > 1:
-        outer_root_context: Context = outer_ctx.new()
-        # NOTE_1:
-        # - Index 0 are the defaults set in BaseContext
-        # - Index 1 is the context generated by `Component.get_context_data`
-        #   of the parent's component
-        # - All later indices (2, 3, ...) are extra layers added by the rendering
-        #   logic (each Node usually adds it's own context layer)
+    # Special case for handling outer context of top-level components when
+    # slots are isolated. In such case, the entire outer context is to be the
+    # outer root ctx.
+    if (
+        outer_ctx
+        and not context.get(_PARENT_COMP_KEY)
+        and app_settings.SLOT_CONTEXT_BEHAVIOR == SlotContextBehavior.ISOLATED
+        and _OUTER_ROOT_CTX_CONTEXT_KEY in context  # <-- Added to avoid breaking tests
```

Comment:
As the comment says, the last line was added to avoid breaking other tests (not quite sure why)